### PR TITLE
feat: respect F_DATAVALUE_ADD user authority

### PR DESCRIPTION
**Pull request by Mohammer5**
Closes TECH-1439 ([link](https://dhis2.atlassian.net/browse/TECH-1439))

* [x] Shows a "locked" info notice box when authorities are missing
* [x] Disabled inputs when authorities are missing
* [x] Disabled (un-)marking for followup when authorities are missing
* [x] Disabled adding/editing comments when authorities are missing
* [x] Disabled settings/deleting min- & max-limits when authorities are missing
* [x] Ensures that `<LoadApp />` always returns something (no implicit `undefined` return)
* [x] Removes `console.log` statement from `use-current-date-string-at-server-timezone.js`

NOTE: @HendrikThePendric commented on the tooltip issue. We've agreed to tackle that separately as it's already tracked in [TECH-1341](https://dhis2.atlassian.net/browse/TECH-1341) and it's not a new issue of this PR.

### CODE REVIEW

File: src/app/load-app.js [hunk 1]
```diff
@@ -9,37 +9,42 @@ import css from './load-app.module.css'
...
+    if (metadata.isError) {
         return (
             <NoticeBox
-                className={css.noticeBoxWrapper}
                 error
+                className={css.noticeBoxWrapper}
                 title={i18n.t('There was a problem loading metadata')}
             >
-                {error?.message}
+                {metadata.error}
```
Comment (HendrikThePendric) [inline src/app/load-app.js:30]: Isn't error and object? And wouldn't we want to render a string?

File: src/app/load-app.js [hunk 2]
```diff
@@ -9,37 +9,42 @@ import css from './load-app.module.css'
...
+    if (!metadata.data) {
+        return (
+            <NoticeBox
+                error
+                className={css.noticeBoxWrapper}
+                title={i18n.t('There was a problem loading metadata')}
+            >
+                {i18n.t('The metadata response is invalid')}
+            </NoticeBox>
+        )
```
Comment (HendrikThePendric) [inline src/app/load-app.js:44]: Why is this here? Does this actually happen? I would have expected that if `metadata.isLoading` and `metadata.isError` are both `false`, that `metadata.data` would always be populated.

File: src/app/load-app.js (hunk 1 above)
Comment (HendrikThePendric) [reply src/app/load-app.js:30]: Also, what about the `userInfo.isError` case? Don't we need to show an error notification for that too? I expect that other parts of the app are relying on the user info to be there.

File: src/data-workspace/data-details-sidebar/basic-information-follow-up-button.js [hunk 3]
```diff
@@ -56,8 +59,21 @@ const FollowUpButton = ({ item }) => {
...
 
+    if (!canUserEditFields) {
+        return (
+            <Tooltip
+                content={i18n.t(
+                    'You do not have the authority to mark this value for follow -up'
+                )}
+            >
+                {markForFollowUpButton}
+            </Tooltip>
```
Comment (HendrikThePendric) [inline src/data-workspace/data-details-sidebar/basic-information-follow-up-button.js:70]: I believe this won't work. The issue is that once a form element (incl. the `button` tag) is `disabled` it stops emitting events. Since the `button` covers the entire tooltip wrapper, the `mousenter` and `mouseleave` events will not fire and the tooltip won't show.

To deal with this type of situation the `Tooltip` from @dhis2/ui supports render-props and you can do something like the last example here: https://codesandbox.io/s/disabled-component-with-tooltip-ubtyk?file=/src/App.js

File: src/data-workspace/data-details-sidebar/comment.js [hunk 4]
```diff
@@ -258,14 +272,17 @@ export default function Comment({ item }) {
...
-                {item.comment ? i18n.t('Edit comment') : i18n.t('Add comment')}
-            </Button>
+            {!canUserEditFields && (
+                <Tooltip
+                    content={i18n.t(
+                        'You do not have the authority to add or edit comments'
+                    )}
+                >
+                    {addEditButton}
+                </Tooltip>
```
Comment (HendrikThePendric) [inline src/data-workspace/data-details-sidebar/comment.js:282]: See my previous comment. Perhaps it makes sense to have a dedicated component for this. I.e. a `ButtonWithTooltip` or something.

File: src/data-workspace/data-details-sidebar/limits-delete-button.js [hunk 5]
```diff
@@ -12,10 +16,25 @@ export default function LimitsDeleteButton({
...
+                content={i18n.t(
+                    'You do not have the authority to delete limits'
+                )}
+            >
+                <Button small secondary disabled>
+                    {label}
+                </Button>
+            </Tooltip>
+        )
+    }
```
Comment (HendrikThePendric) [inline src/data-workspace/data-details-sidebar/limits-delete-button.js:36]: Again, see previous comment

File: src/data-workspace/data-details-sidebar/limits-display.js [hunk 6]
```diff
@@ -12,8 +15,21 @@ import styles from './limits.module.css'
...
 
+    if (!canEditFields) {
+        return (
+            <Tooltip
+                content={i18n.t('You do not have the authority to add limits')}
+            >
+                <Button small primary disabled>
+                    {editButtonLabel}
+                </Button>
+            </Tooltip>
```
Comment (HendrikThePendric) [inline src/data-workspace/data-details-sidebar/limits-display.js:29]: Again. See previous comment.

File: src/data-workspace/data-details-sidebar/no-limits.js [hunk 7]
```diff
@@ -2,15 +2,31 @@ import i18n from '@dhis2/d2-i18n'
...
+        return (
+            <Tooltip
+                content={i18n.t('You do not have the authority to add limits')}
+            >
+                <Button small disabled className={noLimitsStyles.addButton}>
+                    {buttonLabel}
+                </Button>
+            </Tooltip>
+        )
+    }
```
Comment (HendrikThePendric) [inline src/data-workspace/data-details-sidebar/no-limits.js:28]: Repeat

**Review by HendrikThePendric — CHANGES_REQUESTED** [review]
the overall approach is looking good. But I think there is a structural problem with tooltips which wrap a disabled button, and think we might want to handle things slightly differently in `LoadApp`.

**Comment by HendrikThePendric** [comment]
Hadn't realised this, but the issue with the tooltip not working on disabled elements is browser specific. On FireFox things are already working as expected, but not on Chrome.

File: src/data-workspace/data-details-sidebar/basic-information-follow-up-button.js (hunk 3 above)
Comment (Mohammer5) [reply src/data-workspace/data-details-sidebar/basic-information-follow-up-button.js:70]: See note in the PR's description

File: src/data-workspace/data-details-sidebar/limits-delete-button.js (hunk 5 above)
Comment (Mohammer5) [reply src/data-workspace/data-details-sidebar/limits-delete-button.js:36]: See note in the PR's description

File: src/data-workspace/data-details-sidebar/comment.js (hunk 4 above)
Comment (Mohammer5) [reply src/data-workspace/data-details-sidebar/comment.js:282]: See note in the PR's description

File: src/data-workspace/data-details-sidebar/limits-display.js (hunk 6 above)
Comment (Mohammer5) [reply src/data-workspace/data-details-sidebar/limits-display.js:29]: See note in the PR's description

File: src/data-workspace/data-details-sidebar/no-limits.js (hunk 7 above)
Comment (Mohammer5) [reply src/data-workspace/data-details-sidebar/no-limits.js:28]: See note in the PR's description

File: src/app/load-app.js (hunk 2 above)
Comment (Mohammer5) [reply src/app/load-app.js:44]: What I observed in the code prior to my changes is:

* There is no explicit condition for when this is the case, but there's a check for the existence of `data` (`metadata.data` after I changed the code)
* Children won't be rendered if there's no user data or an error.

I suppose in regards to the user info we could add a case similar to the one I added for missing metadata data for the userData as well, because right now there would be just a blank workspace if that happens

File: src/app/load-app.js (hunk 1 above)
Comment (Mohammer5) [reply src/app/load-app.js:30]: See comment below

File: src/app/load-app.js (hunk 2 above)
Comment (Mohammer5) [reply src/app/load-app.js]: @HendrikThePendric and I have agreed that it doesn't make much sense to check for the presence of `.data`, so now all loading and error states have been covered instead

**Review by HendrikThePendric — APPROVED** [review]
I agree we can tackle the tooltip issue in a separate PR, but we do need to make sure it gets implemented during the hard-freeze. To ensure that happens I've moved [TECH-1341](https://dhis2.atlassian.net/browse/TECH-1341) into the [TECH-1356](https://dhis2.atlassian.net/browse/TECH-1356) epic.
